src/EventStore.ClientAPI/Internal/ClusterDnsEndPointDiscoverer.cs

```diff
@@ -109,17 +112,31 @@ internal class ClusterDnsEndPointDiscoverer : IEndPointDiscoverer {
 			return null;
 		}
 
-		private GossipSeed[] GetGossipCandidatesFromDns() {
+		private IGossipSeed[] GetGossipCandidatesFromConfig() {
 			//_log.Debug("ClusterDnsEndPointDiscoverer: GetGossipCandidatesFromDns");
-			GossipSeed[] endpoints;
-			if (_gossipSeeds != null && _gossipSeeds.Length > 0) {
-				endpoints = _gossipSeeds;
+			
+			// Safe in this case.
+			IGossipSeed[] endpoints = _gossipSeeds;
+
+			if ((endpoints?.Length ?? 0) == 0) {
+				if (_compatibilityMode.IsAutoCompatibilityModeEnabled()) {
+					endpoints = new[] {
+						new ClusterDnsSeed(_clusterDns, _managerExternalHttpPort, seedOverTls: true, v20HostHeader: true), // Then HTTPS gossip first (v20 defaults)
+						new ClusterDnsSeed(_clusterDns, _managerExternalHttpPort, seedOverTls: false, v20HostHeader: false), // Try HTTP gossip (v5 defaults)
+					};
+				} else if (_compatibilityMode.IsVersion5CompatibilityModeEnabled()) {
```

Comment:
@YoEight It looks like the v20 and v21 compatibility mode checks should be done here instead of v5 compatibility mode since the user is already using version 5 of the client.